### PR TITLE
fix(buttons): abstract mixins and fix dark states

### PR DESCRIPTION
**Pull request by bfbiggs**

### CODE REVIEW

File: scss/settings/buttons.scss [hunk 1]
```diff
@@ -9,6 +9,7 @@
 @import '../tools/functions/core';
 
 $button__class: 'cui-button';
```
Comment (jlowery457) [inline scss/settings/buttons.scss]: Does the 'cui' prefix need to be a variable for the class names?

File: scss/settings/buttons.scss [hunk 2]
```diff
@@ -46,72 +47,51 @@ $button__background-color: $black-12 !default;
...
 $button--yellow__backround-color--hover: $yellow-hover !default;
-$button--yellow__backround-color--focus: $yellow-base !default;
 $button--yellow__backround-color--active: $yellow-press !default;
-$button--yellow__border-color: null !default;
 
 // Mint button colors
 $button--mint__backround-color: $mint-base !default;
-$button--mint__backround-color--hover: $mint-hover !default;
-$button--mint__backround-color--focus: $mint-base !default;
+$button--mint__backround-color--hover: $mint-hover !default;$button--mint__backround-color--focus: $mint-base !default;
```
Comment (jlowery457) [inline scss/settings/buttons.scss]: Carriage return?

File: scss/tools/mixins/flex.scss [hunk 3]
```diff
@@ -1,6 +1,6 @@
 
-@mixin flex($fd: row, $fw: nowrap, $jc: center, $ac: center, $ai: center) {
-  display: flex;
+@mixin flex($display: flex, $fd: row, $fw: nowrap, $jc: center, $ac: center, $ai: center) {
```
Comment (jlowery457) [inline scss/tools/mixins/flex.scss]: nice.  I was thinking to doing this too.  :)

File: scss/settings/buttons.scss (hunk 2 above)
Comment (bfbiggs) [reply scss/settings/buttons.scss]: Actually a missed deletion but agreed.

File: scss/settings/buttons.scss (hunk 1 above)
Comment (bfbiggs) [reply scss/settings/buttons.scss]: $button__class: '#{$cui}-button';

**Pull request closed without merging**